AGENTS.md

```diff
@@ -0,0 +1,261 @@
+- **dotnet-counters**: Performance counter monitoring tool
+- **Diagnostic libraries**: Client libraries and services for diagnostics
+
+## Build System
+
+### Building the Repository
+
+The repository uses a cross-platform build system:
+
+**Linux/macOS:**
+```bash
+./build.sh
+```
+
+**Windows:**
+```cmd
+Build.cmd
+```
+
+### Build Scripts Location
+- Main build scripts: `build.sh` / `Build.cmd` at repository root
+- Actual build logic: `eng/build.sh` / `eng/Build.cmd`
+- Build configuration: `build.proj`, `Directory.Build.props`, `Directory.Build.targets`
+
+### Common Build Options
+- `--configuration <Debug|Release>`: Build configuration (default: Debug)
+- `--architecture <x64|x86|arm|arm64>`: Target architecture
+- `--restore`: Restore dependencies before building
+- `--build`: Build the repository
+- `--rebuild`: Clean and rebuild
+- `--test`: Run tests after building
+
+## Testing
+
+### Running Tests
+
+**Linux/macOS:**
+```bash
+./test.sh
+```
+
+**Windows:**
+```cmd
+Test.cmd
+```
+
+The test script runs all tests in the repository. Test projects are located in the `src/tests` directory.
+
+### Test Organization
+- Unit tests are typically in `*.Tests` projects
+- Integration tests may be in separate test projects
+- Test helpers are in `Microsoft.Diagnostics.TestHelpers`
+
+## Project Structure
+
+```
+/src
+├── Microsoft.Diagnostics.DebugServices        # Debug service interfaces
+├── Microsoft.Diagnostics.DebugServices.Implementation  # Debug service implementations
+├── Microsoft.Diagnostics.ExtensionCommands    # SOS extension commands
+├── Microsoft.Diagnostics.Monitoring           # Monitoring libraries
+├── Microsoft.Diagnostics.NETCore.Client       # Diagnostic client library
+├── Microsoft.Diagnostics.Repl                 # REPL infrastructure
+├── Microsoft.FileFormats                      # File format parsers
+├── Microsoft.SymbolStore                      # Symbol store implementation
+├── SOS                                        # SOS debugger extension
+├── Tools                                      # Command-line tools (dump, trace, counters, gcdump)
+├── tests                                      # Test projects
+└── shared                                     # Shared native code
+
+/documentation                                  # Documentation files
+/eng                                           # Engineering/build infrastructure
+```
+
+## Coding Standards
+
+### C# Code Style
+
+The repository follows standard .NET coding conventions:
+
+1. **Indentation**: 4 spaces (no tabs)
+2. **Line endings**: LF on Linux/macOS, CRLF on Windows
+3. **Braces**: Opening braces on new lines for types, methods, properties, control blocks
+4. **Naming**:
+   - PascalCase for types, methods, properties, public fields
+   - camelCase for local variables, parameters, private fields
+   - `_camelCase` for private instance fields (with underscore prefix)
+5. **File organization**: One type per file, filename matches type name
```

Comment:
Merge this section and the next - the editor config is part of the coding standard of this repo and is a *must* for c# code.